Updated broken development-guide links

jasonodoom opened the pull request:
# <!-- Use the title to describe PR changes in the imperative mood --> #

## 🗣 Description ##
I updated some broken links in the development guide both in `./dev_envs/mac-env-setup.md` and `onboarding-checklist.md`. 

<!-- Describe the "what" of your changes in detail. -->
<!-- To avoid scope creep, limit changes to a single goal. -->

## 💭 Motivation and context ##

As a new member of the team I stumbled upon these broken links while on-boarding and want to fix them so they're accessible in the future for the next person.
 
<!-- Why is this change required? -->
<!-- What problem does this change solve? How did you solve it? -->
<!-- Mention any related issue(s) here using appropriate keywords such -->
<!-- as "closes" or "resolves" to auto-close them on merge. -->

## ✅ Pre-approval checklist ##

<!-- Remove any of the following that do not apply. -->
<!-- Draft PRs should have one or more unchecked boxes. -->
<!-- If you're unsure about any of these, don't hesitate to ask. -->
<!-- We're here to help! -->

- [X] This PR has an informative and human-readable title.
- [X] Changes are limited to a single goal - *eschew scope creep!*
- [X] All relevant type-of-change labels have been added.
- [X] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [X] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [X] All relevant repo and/or project documentation has been updated to reflect the changes in this PR.

jasonodoom committed:
changed failed VM JIRA link from ncats to ceil

jasonodoom committed:
updated failed wiki links to point to their current location at cool-system-internal

dav3r reviewed (approved):
Thanks for cleaning up those links! 👍 ⭐

jsf9k reviewed (approved):
Thanks for the updates @jasonodoom!  Please note for the future that commit messages should be in the imperative mood, as is mentioned [here](https://cbea.ms/git-commit/#imperative) (linked to from the development guide [here](https://github.com/cisagov/development-guide/#git-github-and-you)).

mcdonnnj commented on dev_envs/mac-env-setup.md:
```diff
@@ -249,14 +249,14 @@ and paste the contents of your clipboard
 There are some additional setup steps to access the
 [COOL (Cloud-Optimized Operations Lab)](https://github.com/cisagov/cool-system/wiki/):
```
We should probably update this to mention that these steps are only necessary if you want to manually configure these vs using the Jamf-pushed configuration.

jasonodoom replied:
added verbatim.

mcdonnnj replied:
Added in [31a9e38](https://github.com/cisagov/development-guide/pull/87/commits/31a9e3882f1ae43807effc191baf76d3c148bbd4).

jasonodoom committed:
Update dev team project board link

jasonodoom committed:
Add note for optional COOL setup steps

mcdonnnj commented on dev_envs/mac-env-setup.md:
```diff
@@ -249,14 +249,19 @@ and paste the contents of your clipboard
 [COOL (Cloud-Optimized Operations Lab)](https://github.com/cisagov/cool-system/wiki/):
 
-1. [Install MIT Kerberos](https://github.com/cisagov/cool-system/wiki/Installing-MIT-Kerberos)
+{% note %}
+**Note:** These steps are only necessary if you want to manually
+configure these vs using the Jamf-pushed configuration.
+{% endnote %}
```
```suggestion
> **Note**
> These steps are only necessary if you want to manually configure these vs
> using the Jamf-pushed configuration.
```

jsf9k commented on dev_envs/mac-env-setup.md:
```diff
@@ -249,14 +249,21 @@ and paste the contents of your clipboard
 
-1. [Install MIT Kerberos](https://github.com/cisagov/cool-system/wiki/Installing-MIT-Kerberos)
+**Note:**
+These steps are only necessary if you want to manually configure these vs
```
Extremely minor suggestion.  "vs" is an abbreviation for "versus", and in American English abbreviations like this always end in a period.  (British English _does not_ use a period if the last letter of the abbreviation is the last letter of the word, which is also why they write "Mr" instead of "Mr.".) 
```suggestion
These steps are only necessary if you want to manually configure these vs.
```

jasonodoom replied:
fair point. fixed!

jsf9k reviewed (commented):
One extremely minor suggestion.

jsf9k commented:
@mcdonnnj, we still need your re-review here.

mcdonnnj commented on dev_envs/mac-env-setup.md:
```diff
@@ -249,14 +249,21 @@ and paste the contents of your clipboard
 [COOL (Cloud-Optimized Operations Lab)](https://github.com/cisagov/cool-system/wiki/):
 
-1. [Install MIT Kerberos](https://github.com/cisagov/cool-system/wiki/Installing-MIT-Kerberos)
+**Note:**
+These steps are only necessary if you want to manually configure these vs.
+using the Jamf-pushed configuration.
```
This is a [specific GitHub-flavored Markdown syntax](https://github.com/orgs/community/discussions/16925) that I suggested. Also I would like consensus on if we prefer `vs` or `vs.` as #79 has the former and this has the latter (pre-suggestion).

```suggestion
> **Note**
> These steps are only necessary if you want to manually configure these vs.
> using the Jamf-pushed configuration.
```

jsf9k replied:
The dot is grammatically correct for American English.

mcdonnnj replied:
Yes I prefer the dot but it was not mentioned in the other PR so I wanted to get a team consensus on preference.

dav3r replied:
Team Dot over here!

jsf9k replied:
I think the three of us makes it an official consensus!

jsf9k replied:
I added the dot to @mcdonnnj's suggested change.

jsf9k replied:
@mcdonnnj - please see commits dc38701 and a1831ea.

jsf9k commented:
BTW @jasonodoom, as a member of @cisagov/team-ois you don't have to fork these repos.  You can just clone them, create a new branch, push it up, and create your PR.

jsf9k committed:
Use the new "Note" syntax where it makes sense

jsf9k committed:
Use the New "Note" syntax in a few more places

jsf9k committed:
Merge branch 'develop' into develop

jsf9k committed:
Remove unnecessarily duplicated line

jsf9k committed:
Remove trailing whitespace

jsf9k commented on dev_envs/mac-env-setup.md:
```diff
@@ -253,16 +258,23 @@ and paste the contents of your clipboard
-1. [Configuring your browser for the COOL](https://github.com/cisagov/cool-system/wiki/Configuring-your-browser-for-the-COOL)
+1. [Configuring Kerberos for the COOL](https://github.com/cisagov/cool-system-internal/blob/master/Configuring-Kerberos-for-the-COOL.md)
+1. [Configuring your browser for the COOL](https://github.com/cisagov/cool-system-internal/blob/master/Configuring-your-browser-for-the-COOL.md)
+1. [Configuring Kerberos for the COOL](https://github.com/cisagov/cool-system-internal/blob/master/Configuring-Kerberos-for-the-COOL.md)
+1. [Configuring your browser for the COOL](https://github.com/cisagov/cool-system-internal/blob/master/Configuring-your-browser-for-the-COOL.md)
```
@jasonodoom - I think the last two bullets are duplicates.  Can you verify and add this change it that's the case?
```suggestion
```

mcdonnnj commented on dev_envs/mac-env-setup.md:
```diff
@@ -217,7 +220,8 @@ git config --global commit.gpgsign true
 Reference: [GitHub: Generating a new SSH key](https://docs.github.com/en/github/authenticating-to-github/generating-a-new-ssh-key-and-adding-it-to-the-ssh-agent#adding-your-ssh-key-to-the-ssh-agent)
 
-Note: use `ed25519` instead of `rsa`
+> **Note**
```
Thoughts on using an alternate display for these since it's more of a mandate than a "just for your informatino" item?

```suggestion
> **Warning**
```

mcdonnnj commented on dev_envs/mac-env-setup.md:
```diff
@@ -236,7 +240,8 @@ ssh-keygen -t ed25519
 Reference: [GitHub: Adding a new SSH key to your GitHub account](https://docs.github.com/en/github/authenticating-to-github/adding-a-new-ssh-key-to-your-github-account)
 
-Note: use `ed25519` instead of `rsa`
+> **Note**
```
Same as the previous suggestion.

```suggestion
> **Warning**
```

jsf9k committed:
Prefer warning

In these two cases the statement is stronger than a "note":  ED25519 keys _must_ be used.

Co-authored-by: Nick <50747025+mcdonnnj@users.noreply.github.com>

jsf9k committed:
Remove duplicate list items

mcdonnnj commented on dev_envs/mac-env-setup.md:
```diff
@@ -253,16 +258,21 @@ and paste the contents of your clipboard
 There are some additional setup steps to access the
 [COOL (Cloud-Optimized Operations Lab)](https://github.com/cisagov/cool-system/wiki/):
```
Since this repo went private the wiki is no longer available.

```suggestion
[COOL (Cloud-Optimized Operations Lab)](https://github.com/cisagov/cool-system/):
```

mcdonnnj commented on onboarding-checklist.md:
```diff
@@ -142,8 +142,8 @@ improved and open as PRs on the [template onboarding checklist].
 [template onboarding checklist]: https://github.com/cisagov/development-guide/blob/master/onboarding-checklist.md
 [NCATS Google Drive folder]: https://drive.google.com/drive/folders/0APw76nbCAmzuUk9PVA
 [Dev Team - Events & OOO calendar]: https://calendar.google.com/calendar/embed?src=beta.dhs.gov_pou7e62ob7h1qadu1m8n4o996g%40group.calendar.google.com&ctz=America%2FNew_York
```
Given the GWE migration should these be updated now since they will soon be broken? There are other Google URLs in here as well I believe.

dav3r replied:
I think any internal links like this should be removed here and just make this a checklist of items that the new Devvie should ask an existing Devvie about.

jsf9k replied:
OK, I will make that change.

jsf9k replied:
Please see commits 11ec021 and 32fcef3 and let me know what you think.

dav3r replied:
Those both look great to me. 👍

jsf9k replied:
@mcdonnnj, can you let me know if we can resolve this conversation or not?  We are also waiting on a re-review from you for this PR.

jsf9k committed:
Update link

The wiki no longer exists since the repo went private.

Co-authored-by: Nick <50747025+mcdonnnj@users.noreply.github.com>

jsf9k committed:
Clean up formatting

jsf9k committed:
Remove references to Trio

Update these references to point to CISA GWE instead, where
appropriate.

Also remove references to JIRA and the dev team GH project board,
since the former is going away and the latter is of limited use.

Glom together sections on dev and non-dev team members.  This document
is for dev team members and QA team members only, so it doesn't make
sense to break out a separate section for non-members.

dav3r reviewed (approved):
👍 👍

dv4harr10 reviewed (commented):
Goto code/development-guide/onboarding-checklist.md line 92 link https://groups.google.com/a/beta.dhs.gov/d/forum/all returns error message 'Content unavailable. Click here to try again'. After click, returns Google 403 error. We're sorry, but you do not have access to this page.

dv4harr10 commented:
The same error results are noticed for links on line 93 thru line 99: - [ ] [CISA COOL Administrators](https://groups.google.com/a/trio.dhs.gov/forum/#!forum/cisa-cool-group)
  - [ ] [cisa-cool-account-support](https://groups.google.com/a/trio.dhs.gov/d/forum/cisa-cool-account-support-group)
  - [ ] [cisa-cool-users](https://groups.google.com/a/trio.dhs.gov/d/forum/cisa-cool-users-group)
  - [ ] [cisagov-github](https://groups.google.com/a/trio.dhs.gov/forum/#!forum/cisagov-github-group)
  - [ ] [NCATS](https://groups.google.com/a/beta.dhs.gov/d/forum/ncats)
  - [ ] [NCATS AWS Support](https://groups.google.com/a/beta.dhs.gov/d/forum/ncats-aws-support)
  - [ ] [NCATS Infrastructure Development](https://groups.google.com/a/beta.dhs.gov/d/forum/ncats-dev)

dav3r commented:
> Goto code/development-guide/onboarding-checklist.md line 92 link https://groups.google.com/a/beta.dhs.gov/d/forum/all returns error message 'Content unavailable. Click here to try again'. After click, returns Google 403 error. We're sorry, but you do not have access to this page.

> The same error results are noticed for links on line 93 thru line 99..

@dv4harr10 All of those links are internal and you don't have access to them, but more importantly, they were removed via commit https://github.com/cisagov/development-guide/pull/87/commits/32fcef357d26795594a400f688c985d1545f8658, so there is no problem here.

jmorrowomni commented:
> The same error results are noticed for links on line 93 thru line 99: - [ ] [CISA COOL Administrators](https://groups.google.com/a/trio.dhs.gov/forum/#!forum/cisa-cool-group)
> 
> * [ ]  [cisa-cool-account-support](https://groups.google.com/a/trio.dhs.gov/d/forum/cisa-cool-account-support-group)
> * [ ]  [cisa-cool-users](https://groups.google.com/a/trio.dhs.gov/d/forum/cisa-cool-users-group)
> * [ ]  [cisagov-github](https://groups.google.com/a/trio.dhs.gov/forum/#!forum/cisagov-github-group)
> * [ ]  [NCATS](https://groups.google.com/a/beta.dhs.gov/d/forum/ncats)
> * [ ]  [NCATS AWS Support](https://groups.google.com/a/beta.dhs.gov/d/forum/ncats-aws-support)
> * [ ]  [NCATS Infrastructure Development](https://groups.google.com/a/beta.dhs.gov/d/forum/ncats-dev)

@dv4harr10 That entire section has been removed in commit 32fcef3

dv4harr10 commented:
Thanks for the updates. Also, in the onboarding-checklist.md file line 145 link returns the message ‘This site can’t be reached’.

@ https://jira.ncats.cyber.dhs.gov/secure/Dashboard.jspa

dav3r commented:
> Thanks for the updates. Also, in the onboarding-checklist.md file line 145 link returns the message ‘This site can’t be reached’.
> 
> @ https://jira.ncats.cyber.dhs.gov/secure/Dashboard.jspa

@dv4harr10 That link was also removed in https://github.com/cisagov/development-guide/pull/87/commits/32fcef357d26795594a400f688c985d1545f8658.  Please ensure that you are looking at the [current version of this pull request](https://github.com/cisagov/development-guide/pull/87/files) before you note any additional issues.

I also encourage you to use GitHub PR review comments to indicate exactly which lines of code you are referring to (see [here](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/reviewing-proposed-changes-in-a-pull-request) for more info).

jsf9k committed:
Merge branch 'develop' into develop

dav3r reviewed (approved):
Approval intensifies!

mcdonnnj commented on onboarding-checklist.md:
```diff
@@ -55,58 +56,41 @@ make sure you're assigned to this issue.
+      before we can give you access to any systems, including CISA GWE
+      (Google Workspace Enterprise).
 
 ### Microsoft Teams chat ###
```
This section should be updated to remove the links since we purged other internal links in this PR.

jsf9k replied:
Please see commit c066db4.

mcdonnnj replied:
@jsf9k You missed the leading `[` for the link on the line before. Do you want to roll that into the commit you made and force-push?

jsf9k replied:
Please see commit 58d3788.

jsf9k committed:
Remove link to internal Sharepoint training site

mcdonnnj commented on onboarding-checklist.md:
```diff
@@ -55,58 +56,40 @@ make sure you're assigned to this issue.
+Once you're set up on Microsoft Teams, we recommend you join the
+following:
 
 - [ ] [CISA](https://teams.microsoft.com/l/team/19%3aa40fd338ea7a45de994a7c7ce9671fd4%40thread.skype/conversations?groupId=8c6c0b00-7244-46e1-bef9-76ce2a1906f5&tenantId=3ccde76c-946d-4a12-bb7a-fc9d0842354a)
 - [ ] [CSD All Staff](https://teams.microsoft.com/l/team/19%3a65f1c4b58edf4246995ae3fa732767ee%40thread.skype/conversations?groupId=6dfc3f23-3a49-44ce-ad37-6c7a9d8fb0aa&tenantId=3ccde76c-946d-4a12-bb7a-fc9d0842354a)
 - [ ] [Vulnerability Management](https://teams.microsoft.com/l/team/19%3a2161749c8eb94adeb31480e298f3b532%40thread.skype/conversations?groupId=61e939a6-2165-4085-a26e-decae97d1471&tenantId=3ccde76c-946d-4a12-bb7a-fc9d0842354a)
```
Are we considering these internal links to be purged as well? It seemed like were just wholesale axing anything that isn't on GitHub.

jsf9k replied:
@cisagov/team-ois, what do you think?

dav3r replied:
I'm fine getting rid of these since we don't really use them.

jsf9k replied:
@mcdonnnj, please see commit 30503ef.

jsf9k committed:
Remove section about Microsoft Teams chats

We don't actually use any of these chats, so there is no reason to
mention them.

Co-authored-by: Nicholas McDonnell <50747025+mcdonnnj@users.noreply.github.com>

mcdonnnj commented on onboarding-checklist.md:
The links for the `CISA COOL AWS Accounts` and `Performance Plans` documents are both internal Google links so they should probably be removed. Additionally the wiki is currently off in [cisagov/cool-system](https://github.com/cisagov/cool-system) due to the previous situation around data exposure. Should we either 1) turn on the wiki in that repository or 2) rename the URL's link text to accurately reflect what you'd land on if you visit it?

dav3r replied:
We can't turn on the wiki in cool-system since it's a private repo and we don't pay for that feature.  Also, the COOL Wiki markdown docs are no longer in cool-system anyway, so we might as well just remove that link or else point to the correct private repo.

jsf9k replied:
Please see commits d6b24a5 and 6cf9f5d.

dav3r replied:
Regarding https://github.com/cisagov/development-guide/commit/6cf9f5d1b28d8e85054cb639c222ce40cbf09107,  I think @mcdonnnj was saying that we shouldn't name the link "COOL Wiki" if it's not pointing to an actual wiki anymore- just a wiki-like collection of Markdown files.  I don't have a good suggestion for what we should rename this link to - got any ideas?

jsf9k replied:
Please see commit f96a50d.

dav3r replied:
Works for me. 👍

mcdonnnj replied:
@dav3r Regarding turning on the wiki: I forgot both `cool-system` repos were private (because that makes sense...).

jsf9k committed:
Remove two more internal links

The links to the COOL accounts and performance plans spreadsheets
point to internal documents, so we are removing them.

Co-authored-by: Nicholas McDonnell <50747025+mcdonnnj@users.noreply.github.com>

jsf9k committed:
Update COOL wiki link to point to cisagov/cool-system-internal

Co-authored-by: Nicholas McDonnell <50747025+mcdonnnj@users.noreply.github.com>

jsf9k committed:
Describe why the COOL wiki is now just a collection of MD pages

Co-authored-by: Nicholas McDonnell <50747025+mcdonnnj@users.noreply.github.com>
Co-authored-by: David Redmin <david.redmin@gwe.cisa.dhs.gov>
